ext/GraphvizVisualization.jl

```diff
@@ -0,0 +1,667 @@
+module GraphvizVisualization
+
+export generate_dot, show_gv, dot_string_to_pdf, SimpleIteration, BFSTraversal
+
+using GraphPPL, MetaGraphsNext, GraphViz
+using GraphPPL.MetaGraphsNext
+import MetaGraphsNext: nv
+
+"""
+This abstract type represents a node traversal strategy for use with the `generate_dot` function.
+
+This abstract type is used to define various strategies for traversing nodes in the graph when generating a DOT representation. 
+Each concrete subtype specifies a different traversal approach, which is selected by Julia's multiple dispatch system 
+when calling `generate_dot`.
+
+Concrete subtypes:
+- `SimpleIteration`: Represents a simple iteration of the graph's vertex/node set.
+- `BFSTraversal`: Represents a breadth-first search traversal strategy, from the initially created node.
+
+These types afford a trade-off between a relatively fast and a relatively 'principled' iteration strategy (respectfully).
+"""
+abstract type TraversalStrategy end
```

Comment:
This type should be available when we load GraphPPL and GraphViz, I don't really see a quick fix for this but I'm sure we can find something out